status/condition_set.go

```diff
@@ -237,3 +239,58 @@ func (c ConditionSet) findUnhealthyDependents() []Condition {
 	})
 	return conditions
 }
+
+// Unstructured is an embedded struct type that embeds the upstream type
+type Unstructured struct {
```

Comment:
```suggestion
type objectImpl struct {
  conditions []metav1.Condition
  }
```

Consider defining your structured type as an implementation of the `status.Obejct` interface. This object stores `[]metav1.Condition`s inside of it. It also doesn't need to be exported, since you will just use it when you are trying to convert an object from `Unstructured` into something that implements `status.Object`